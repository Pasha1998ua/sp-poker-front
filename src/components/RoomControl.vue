<template>
  <div class="control-block">
    <button @click="joinRoom">Join</button>
    <button @click="leaveRoom">Leave</button>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import SocketHelper from '@/helpers/socketHelper'

const SOCKET_HELPER = SocketHelper.getInstance()

export default defineComponent({
  name: 'MainPage',
  data () {
    return {}
  },
  methods: {
    joinRoom () {
      SOCKET_HELPER.joinRoom('PavloKey')
    },
    leaveRoom () {
      SOCKET_HELPER.leaveRoom('PavloKey')
    }
  }
})
</script>

<style scoped lang="scss">
</style>
