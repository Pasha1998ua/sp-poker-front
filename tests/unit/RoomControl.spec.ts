import { shallowMount } from '@vue/test-utils'
import RoomControl from '@/components/RoomControl.vue'

describe('RoomControl.vue', () => {
  it('renders page correctly', () => {
    const wrapper = shallowMount(RoomControl)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
