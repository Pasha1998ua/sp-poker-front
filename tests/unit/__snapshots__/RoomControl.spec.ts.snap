// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomControl.vue renders page correctly 1`] = `<div class="control-block"><button>Join</button><button>Leave</button></div>`;
